Updates to GEM-CSC integrated local trigger for Phase2

dildick opened the pull request:
This pull request is an intermediate step towards fully updating the GEM-CSC local triggers for Phase2. It fixes the most pressing issues in the ME11 and ME21 motherboards, namely the construction of the ALCT-GEM type LCTs. The code is far from cleaned up yet. I plan to do that in a final pull request in this series. 

@tahuang1991

makortel committed:
Update campaigns

makortel committed:
Replace file numbering with proper naming

makortel committed:
Separate dxy/dz vs PV and BS, add zoomed plots to vs PV

makortel committed:
Add support for B-hadron plots

makortel committed:
Improve PU sample matching

makortel committed:
Update classifiers in trackingPlots

makortel committed:
Add missing old-style selectors to trackingPlots

makortel committed:
Plot certain vertex plots also for noPU samples

makortel committed:
Adjust axis limits

makortel committed:
Remove highPurity summary production for conversion and gsf

To not add empty columns to summary page as the HP summaries do not
axist for these.

makortel committed:
Add summary tables for pT>0.9 tracks

makortel committed:
Add HTML tables for seeding layer sets

makortel committed:
Add support for multiple pixelPair seeds, and sorted firstStepPrimaryVertices

makortel committed:
Example on MTV customization and plotting for custom track+TrackingParticle selections

makortel committed:
Update timing plots to the updated FastTimerService

makortel committed:
A way to infer iteration order from DQM file for the timing plots

hroskes committed:
hp_alignments hp_deformations

makortel committed:
Create the plots directly in the final directory instead of PWD and move

mmusich committed:
make the vector shrinkage work

mmusich committed:
up

mmusich committed:
towards modularity

mmusich committed:
improve booking of histograms

mmusich committed:
further steps towards scalability

mmusich committed:
furthers steps towards modularity

mmusich committed:
put on par with clean-up branch

mmusich committed:
move some methods to helper class

mmusich committed:
improve book-keeping of residual plot ranges

mmusich committed:
post-merge fixes

mmusich committed:
hardwire binning to plot type

makortel committed:
Fix in mergeBinLabels

mmusich committed:
clean-up

mmusich committed:
use auto-generated bins

mmusich committed:
move more common methods to helpers

cmsbuild commented:
A new Pull Request was created by  @dildick (Sven Dildick) for master.

It involves the following packages:

L1Trigger/CSCTriggerPrimitives


@cmsbuild, @rekovic, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@valuev, @ptcox, @kreczko, @Martin-Grunewald this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 2b33b8d8e8ed2030fc444039f47883e83e30e5f3


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/7b1c63d9e3a493d2602cbc23157526aaac79017f
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21536/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21536/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21536/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/Utilities/StaticAnalyzers/src/ArgSizeChecker.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/Utilities/StaticAnalyzers/src/ConstCastChecker.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/Utilities/StaticAnalyzers/src/ConstCastAwayChecker.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/Utilities/StaticAnalyzers/src/edmChecker.cpp 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.cc: In constructor 'CSCGEMMotherboard::CSCGEMMotherboard(unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, const edm::ParameterSet&)':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.cc:19:87: error: no matching function for call to 'GEMCoPadProcessor::GEMCoPadProcessor(unsigned int&, unsigned int&, int, unsigned int&, const edm::ParameterSet&)'
   coPadProcessor.reset( new GEMCoPadProcessor(endcap, station, 1, chamber, coPadParams) );
                                                                                       ^
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.h:5:0,
                 from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.cc:1:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/GEMCoPadProcessor.h:25:3: note: candidate: GEMCoPadProcessor::GEMCoPadProcessor()
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 64 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/CSCCathodeLCTProcessor.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/CSCMotherboardME3141RPC.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/CSCAnodeLCTProcessor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/plugins/SealModule.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesProducer.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.cc:19:29: error: no matching constructor for initialization of 'GEMCoPadProcessor'
  coPadProcessor.reset( new GEMCoPadProcessor(endcap, station, 1, chamber, coPadParams) );
                            ^                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/L1Trigger/CSCTriggerPrimitives/src/GEMCoPadProcessor.h:21:3: note: candidate constructor not viable: requires 4 arguments, but 5 were provided
  GEMCoPadProcessor(unsigned endcap, unsigned station, unsigned chamber,
  ^
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/7b1c63d9e3a493d2602cbc23157526aaac79017f
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21536/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21536/git-merge-result

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
Pull request #19778 was updated. @cmsbuild, @rekovic, @mulhearn, @davidlange6 can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: c9b4edb1ea1854bcadb92b8899f87aeba62ec38a


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/7b1c63d9e3a493d2602cbc23157526aaac79017f
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21541/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21541/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21541/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
20034.0 step2
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>20434.0 step2
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/7b1c63d9e3a493d2602cbc23157526aaac79017f
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21541/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21541/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
Pull request #19778 was updated. @cmsbuild, @rekovic, @mulhearn, @davidlange6 can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: e6311401f3e53e83c7f91b7c9c4a63cd4fbcb003
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21545/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/f56dad19aae2b1e9c39069555e360e4ea6730bcc
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21545/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21545/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21545/summary.html

Comparison Summary:
 - You potentially added 2709 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2025409
 - DQMHistoTests: Total failures: 29863
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1995380
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

rekovic commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

mmusich committed:
use auto-generated bins everywhere

mmusich committed:
more automation and clean-up

mmusich committed:
add sanity check histogram on the number of ladders in the geometry

mmusich committed:
Update the plotting macro to make Ladder and module number plots

mmusich committed:
graphical improvements in plotting macros

davidlange6 commented on L1Trigger/CSCTriggerPrimitives/src/CSCAnodeLCTProcessor.cc:
```diff
@@ -266,12 +266,12 @@ CSCAnodeLCTProcessor::CSCAnodeLCTProcessor(unsigned endcap, unsigned station,
   // run the ALCT processor for the Phase-II ME2/1 integrated local trigger
-  runME21ILT_ = conf.existsAs<bool>("runME21ILT")?
-    conf.getParameter<bool>("runME21ILT"):false;
+  runME21ILT_ = comm.existsAs<bool>("runME21ILT")?
```
hi @dildick - why are the existsAs needed?

dildick replied:
That parameter is not there in Run2, and Run3 WFs

davidlange6 commented on L1Trigger/CSCTriggerPrimitives/src/CSCMotherboardME11GEM.cc:
```diff
@@ -360,6 +360,10 @@ void CSCMotherboardME11GEM::run(const CSCWireDigiCollection* wiredc,
   clctV1a = clct1a->run(compdc); // run cathodeLCT in ME1/a
+
+  // if there are no ALCTs and no CLCTs, it does not make sense to run this TMB
+  if (alctV.size()==0 and (clctV1b.size()==0 or clctV1a.size()==0)) return;
```
please use alctV.empty() instead

davidlange6 commented on L1Trigger/CSCTriggerPrimitives/src/CSCMotherboardME11GEM.cc:
```diff
@@ -406,47 +410,208 @@ void CSCMotherboardME11GEM::run(const CSCWireDigiCollection* wiredc,
 
+    const int nGEMPads(randRoll->npads());
+    for (int i = 1; i<= nGEMPads; ++i){
+      const LocalPoint lpGEM(randRoll->centreOfPad(i));
```
can this be a const & instead?

davidlange6 commented on L1Trigger/CSCTriggerPrimitives/src/CSCMotherboardME11GEM.cc:
```diff
@@ -406,47 +410,208 @@ void CSCMotherboardME11GEM::run(const CSCWireDigiCollection* wiredc,
+      temp.erase(std::unique(temp.begin(),temp.end()),temp.end());
+      // keep only the middle two or middle element
+      std::set<int> temp2;
+      if (temp.size()==2){
```
please use a switch  or at least if elses

davidlange6 commented on L1Trigger/CSCTriggerPrimitives/src/CSCMotherboardME11GEM.cc:
```diff
@@ -406,47 +410,208 @@ void CSCMotherboardME11GEM::run(const CSCWireDigiCollection* wiredc,
+        temp2.insert(temp[temp.size()/2]);
+      }
+
+      if (temp.size()==0) {
```
please define a helper function

davidlange6 commented on L1Trigger/CSCTriggerPrimitives/src/CSCMotherboardME11GEM.cc:
```diff
@@ -574,52 +752,62 @@ void CSCMotherboardME11GEM::run(const CSCWireDigiCollection* wiredc,
-          // find the best matching copad - first one 
-          auto copads(matchingGEMPads(alct->bestALCT[bx_alct], coPads_[bx_gem], ME1B, true));             
+
+	  std::cout <<"ALCT-to-GEM matching in ME1b, bx_gem "<< bx_gem << std::endl;
```
should only happen for debug enabled?

davidlange6 commented on L1Trigger/CSCTriggerPrimitives/src/CSCMotherboardME11GEM.cc:
```diff
@@ -1120,7 +1371,7 @@ std::vector<CSCCorrelatedLCTDigi> CSCMotherboardME11GEM::sortLCTsByQuality(enum
           for ( ; itp != LCTs_tmp1.end(); itp++)
             {
-              if (me==ME1A and it1a != LCTs1a.end() and *itp==*it1a ) 
+              if (me==ME1A and it1a != LCTs1a.end() and *itp==*it1a )
```
for itp and it1a - can you compare the indices rather than the payloads?

dildick replied:
the collections are sorted differently. I don't think you can use the indices.

davidlange6 commented on L1Trigger/CSCTriggerPrimitives/src/CSCMotherboardME21GEM.cc:
```diff
@@ -242,45 +269,99 @@ CSCMotherboardME21GEM::run(const CSCWireDigiCollection* wiredc,
+        cscHsToGemPad_[158] = std::make_pair(1,1);
+        cscHsToGemPad_[159] = std::make_pair(1,1);
       }
+      // // get unique values
```
remove the commented code

davidlange6 commented on L1Trigger/CSCTriggerPrimitives/src/CSCMotherboardME21GEM.cc:
```diff
@@ -242,45 +269,99 @@ CSCMotherboardME21GEM::run(const CSCWireDigiCollection* wiredc,
+    retrieveGEMCoPads();
+  }
+
+  std::cout << "Available copads " << std::endl;
```
message logger?

cmsbuild commented:
Pull request #19778 was updated. @cmsbuild, @rekovic, @mulhearn, @davidlange6 can you please check and sign again.

dildick commented:
@davidlange6 I implemented some suggestions already

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 06cf95795200c67519a99b49801047c630703653


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/97d9ccda1cab3983b393cbc83487fb038fb791a5
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21648/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21648/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21648/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
20034.0 step2
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>20434.0 step2
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/97d9ccda1cab3983b393cbc83487fb038fb791a5
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21648/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21648/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
Pull request #19778 was updated. @cmsbuild, @rekovic, @mulhearn, @davidlange6 can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 0f5697a2a67de281e749efd4c90c650102120564
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21650/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/97d9ccda1cab3983b393cbc83487fb038fb791a5
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21650/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21650/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19778/21650/summary.html

Comparison Summary:
 - You potentially added 1774 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2346597
 - DQMHistoTests: Total failures: 29993
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2316438
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

makortel committed:
Add built tracks back to trackingOnly MTV

felicepantaleo committed:
distance filter included

felicepantaleo committed:
activating hit energy filter

ianna committed:
DetectorDescription Clang check modernize-use-equals-delete

dildick commented:
@davidlange6 Do you have more comments on this pull request?

mmusich committed:
roothistmatrix in lieu of rootgraphics

dildick commented:
@rekovic do you have any comments on this PR?

ianna committed:
DetectorDescrition and Geometry Clang readability-container-size-empty

felicepantaleo committed:
Merged HGCal_Realistic_simclusters_PR_master from repository felicepantaleo

felicepantaleo committed:
removing white space

felicepantaleo committed:
removing white space

Commit:
gcc7 compilation warning(s) fix(es) in RecoEgamma/EgammaIsolationAlgos package

cmsbuild committed:
Merge pull request #20017 from fwyzard/DQMGenericClient_add_back_protection

Add back a protection for division by 0 to DQMGenericClient

cmsbuild committed:
Merge pull request #20005 from fioriNTU/RearrangingPlotsPhase1

rearranging some plots saving bins

cmsbuild committed:
Merge pull request #20004 from cms-tsg-storm/hltGetConfiguration_fix_93X

 Minor bug fix in hltGetConfiguration - 93X

cmsbuild committed:
Merge pull request #19989 from nsmith-/remove_StableParameters_93x

Remove uGT StableParameters_cff

cmsbuild committed:
Merge pull request #19970 from bsunanda/Run2-alca87

Run2-alca87 Update the macros needed for IsoTrack Calibration

cmsbuild committed:
Merge pull request #19117 from fwyzard/hltGetConfiguration_remove_offline_option

hltGetConfiguration: remove --online/--offline option

kpedro88 committed:
only use hcal endcap ids

cmsbuild committed:
Merge pull request #19984 from Dr15Jones/removeOldFileInPath2

Remove use of old FileInPath class

Commit:
Change uint to unsigned int, and merged ifs

thomreis committed:
Make efficiency histogram binning configurable from config file.

mandrenguyen committed:
Fix for mixing workflows that use playback

Commit:
gcc7 compilation warning fix in RecoMuon/MuonIsolation package

kpedro88 committed:
fix upgrade input tags for HCAL TPs w/ premixing

wddgit committed:
New module which filters on logical expressions of Paths

Commit:
Modify the module name and also generalize it for a number of trigger objects

cmsbuild committed:
Merge pull request #20029 from kpedro88/FixPremixHcalTP

fix upgrade input tags for HCAL TPs w/ premixing

cmsbuild committed:
Merge pull request #20015 from Martin-Grunewald/Fix_hltIntegrationTests_93X

Fix hltIntegrationTests (93X)

cmsbuild committed:
Merge pull request #19974 from ianna/use-graph-from-utils

Use Graph class from DataFormats/Math Package

cmsbuild committed:
Merge pull request #19997 from kpedro88/SimClusNan93X

Avoid NaNs in SimClustering

cmsbuild committed:
Merge pull request #19982 from mohsinwaseem/UpdateHistos

Added missing efficiency histograms.

ianna committed:
mergea Please enter the commit message for your changes. Lines starting

cmsbuild committed:
Merge pull request #20022 from mrodozov/cmp-wngs-fix-gcc7-RecoMuon/MuonIsolation

gcc7 compilation warning fix in RecoMuon/MuonIsolation package

cmsbuild committed:
Merge pull request #19878 from cms-btv-pog/fixValidationPlots_from-CMSSW_9_3_0_pre2

B-tag validation/offline DQM: fix efficiency plots on data & eta ranges

ianna committed:
remove conflict

mmusich committed:
make cut on track multiplicity in SiStrip Hit efficiency configurable

VinInn committed:
remove useless histos

felicepantaleo committed:
using range mask and range match for excluding ghost detids, slava s suggestion

felicepantaleo committed:
changing description

mandrenguyen committed:
removed unnecessary import

smuzaffar committed:
added missing boost dependency needed by Utilities/General/src/FileInPath.cc

this needs to go in as soon as possible

smuzaffar committed:
Merge pull request #20033 from cms-sw/smuzaffar-patch-2

added missing boost dependency needed by Utilities/General/src/FileIn…

mmusich committed:
Update SiStripHitEff_cff.py

Martin-Grunewald committed:
take frozen menu from /frozen ConfDB location

Commit:
Change switch-case to if's

rovere committed:
Make HGCal photons EE

rovere committed:
Monitor HGcal Photons

HGCal photons do not have regular isolation variables
defined. A simpler selection is made using only Et and eta
and pretend those photons are "isolated". This simplified
selection is only activated in the HGCal phaseII scenario.
Also, the invariant mass plot has bin properly partitioned
in barrel-barrel, endcap-endcap and endcap-barrel. This is
true for all cases, not only for the HGCal Phase era.

Martin-Grunewald committed:
fix Fake/1/2 menus

ianna committed:
clang-tidy check readability-redundant-string-cstr

davidlange6 committed:
Merge pull request #20021 from mrodozov/cmp-wngs-fix-rcgm-egmisltnalgs-gcc7

gcc7 compilation warning fix in RecoEgamma/EgammaIsolationAlgo pkg

davidlange6 committed:
Merge pull request #19975 from ianna/geom-clang-modernize-use-emplace

Geometry Clang Check modernize-use-emplace

VinInn committed:
fix configs

Dr15Jones committed:
Removed old FileInPath class

The old FileInPath class is no longer used by any code in CMSSW

Commit:
Add a range in BX

Commit:
Add a HLT filter for Laser events in HCAL

rovere committed:
Review 1.

Commit:
Add the line for makeHLTFilterDescription

Commit:
Changed to EDFilter

wddgit committed:
Improvements in response to PR comments

dildick commented:
@mulhearn @rekovic could you please review and sign?

cmsbuild committed:
Merge pull request #20042 from ianna/dd-readability-redundant-string-cstr

DetectorDescription Clang-tidy Check readability-redundant-string-cstr

cmsbuild committed:
Merge pull request #20034 from Martin-Grunewald/FrozenFromFrozen93X

Take frozen HLT menu from /frozen ConfDB location (93X)

cmsbuild committed:
Merge pull request #20028 from wddgit/pathStatusFilter

New module which filters on logical expressions of Paths

cmsbuild committed:
Merge pull request #20050 from slava77/CMSSW_9_3_0_pre3/sign854/stripMultiplicity

attempt to fix SiStrip DQM reproducibility issue: allign members common to root and full c++

cmsbuild committed:
Merge pull request #17859 from knash/HFchanges

Spike cleaner fix

mmusich committed:
make cut effect same as before

fioriNTU committed:
Upgrade of the SingleModuleOccupancy script

cmsbuild committed:
Merge pull request #20047 from Dr15Jones/removeUtilGenFileInPath

Removed old FileInPath class

vanbesien committed:
Fixed irrational range and spaces in histo labels

rovere committed:
Review 2.

sikler committed:
changed names to TrackingHLTBeamspotStream

hroskes committed:
hp works whether or not deformations were included, and sanity check on conditions

Martin-Grunewald committed:
Handling of @currentProcess

Martin-Grunewald committed:
Add frozen V2.2 and work toward V3

cmsbuild committed:
Merge pull request #20068 from Martin-Grunewald/93XcurrentProcess

Handling of @currentProcess at HLT (93X)

cmsbuild committed:
Merge pull request #20025 from thomreis/l1tmuon_offline_dqm_variable_binning_fix_exception

Make L1TMuon offline DQM histogram binning configurable from config file

cmsbuild committed:
Merge pull request #19972 from natalia-korneeva/btag-paths-modification

Btag paths modification

cmsbuild committed:
Merge pull request #19959 from mrodozov/cmp-wngs-fix-dqm-ecmntrdbmdl-gcc7

gcc7 compilation warning(s) fix(es) in DQM/EcalMonitorDbModule

cmsbuild committed:
Merge pull request #19840 from stahlleiton/DQMOnline_L1TMuon_QTsTune_v20170718_Master

OnlineDQM : Tune L1T Muon QTs v2

davidlange6 committed:
Merge pull request #20070 from cms-tsg-storm/93XHLTmenuV2p2

Add frozen V2.2 and work toward V3 HLT menu (93X)

davidlange6 committed:
Merge pull request #19996 from mrodozov/cmp-wngs-fix-l1trg-glbtrganlzr-gcc7

gcc7 compilation warning(s) fix(es) in L1Trigger/GlobalTriggerAnalyzer

cmsbuild committed:
Merge pull request #19393 from alesaggio/removeSiStripSubDetId_CalibtrackerSiStrip

Deprecation of T**DetId in CalibTracker

Commit:
gcc7 compilation warning(s) fix(es) in L1Trigger/RegionalCaloTrigger

Commit:
gcc7 compilation warning fix in EventFilter/CSCTFRawToDigi

DryRun committed:
Use all samples if samples < ncapid

vanbesien committed:
Updating GT for online (v7) and express (v6)

Commit:
remove 'accept' varaible

cmsbuild committed:
Merge pull request #20053 from fioriNTU/UpgradeOccupancyScript

Upgrade of the SingleModuleOccupancy script

cmsbuild committed:
Merge pull request #19571 from felicepantaleo/HGCal_Realistic_simclusters_PR_master

Phase2 - HGC - Realistic SimClusters

cmsbuild committed:
Merge pull request #18521 from sabasehrish/ss_EF_ESDigiToRaw_ns

converted to global ESDigiToRaw

Commit:
Update the MIP calibration tool for HCAL

cmsbuild committed:
Merge pull request #20008 from felicepantaleo/HGCal_remove_ghost_detids_930pre3

HGCal - Remove ghost detids

cmsbuild committed:
Merge pull request #19927 from mandrenguyen/pfRelinkingHI9XY

Add PF relinking for the HI reco sequence

cmsbuild committed:
Merge pull request #19925 from spandeyehep/offline_PF_calib_for_MC-v2

Offline PF hadron calibration for MC-v2

vanbesien committed:
Additional update to the express GT for 9_2_8

cmsbuild committed:
Merge pull request #20023 from mrodozov/cmp-wngs-fix-gcc7-EventFilter/CSCTFRawToDigi

gcc7 compilation warning fix in EventFilter/CSCTFRawToDigi

cmsbuild committed:
Merge pull request #20014 from sikler/more_realistic_origin_12cm_centered_93X

more realistic cylinder of origin, halflength = 12cm, centered, 93X

cmsbuild committed:
Merge pull request #18322 from sabasehrish/ss_threadsafe_EcalDigiToRaw

Ss threadsafe ecal digi to raw

mmusich committed:
Implement comments from code review

cmsbuild committed:
Merge pull request #19574 from bsunanda/Run2-hcx138

Run2-hcx138 Add provision for making selection on multiple L1 objects

Commit:
Change *good* to *all* for var names of HF

Commit:
Comment on ieta,iphi values used

mmusich committed:
Revert "Implement comments from code review"

This reverts commit 1bf7579266e5d8340252aa9343abf6df13efa1a4.

mmusich committed:
use GetQuantiles in median calculation and use std::string in favor of TString

mmusich committed:
allow correct fetching of extrema in PV Validation plotting macro even in case of merged output files

Commit:
for neutral exotic WIMPs - send their daughters to Geant4 instead of the exotic itself

bbockelm committed:
Re-enable trigger cache.

This re-enables the trigger cache, which provides a bit more than
an order-magnitude speedup for edmCopyPickEvent over high-latency.

cmsbuild committed:
Merge pull request #19637 from makortel/clusterTPAssociationSort

Fix cluster-to-TrackingParticle associations for duplicate entries

cmsbuild committed:
Merge pull request #19795 from kreczko/kreczko/issue/19590

[93X] Improvements to L1 Offline DQM - part 1 (issue #19590)

cmsbuild committed:
Merge pull request #19797 from kreczko/kreczko/issue/19593

[93X] Improvements to L1 Offline DQM - part 3 (issue #19593)

cmsbuild committed:
Merge pull request #19839 from mmusich/93X_PVrehauling_v1

[93X] Streamline Alignment PV Validation tool

cmsbuild committed:
Merge pull request #19889 from kpedro88/Run3-WF1-v4

Introduce Run3 (post-LS2) workflow

cmsbuild committed:
Merge pull request #20044 from VinInn/TkDQMEff93

Cleanup Efficiency plot for Tracking DQM

cmsbuild committed:
Merge pull request #20049 from bsunanda/Run2-hcx139

Run2-hcx139 Add a HLT filter for Laser events in HCAL

cmsbuild committed:
Merge pull request #20052 from mmusich/93X_makeHitEffCutConfigurable

[93X] make cut on number of tracks in Strip Hit Efficiency code configurable

cmsbuild committed:
Merge pull request #20084 from bbockelm/reenable_trigger_cache_93x

Re-enable trigger cache.

Martin-Grunewald committed:
More L1T compat work for TSG use

Martin-Grunewald committed:
More L1T compat work for TSG use

cmsbuild committed:
Merge pull request #20061 from hroskes/hp_validation

Load deformations from HipPy alignments in alignment validation tool

cmsbuild committed:
Merge pull request #20074 from vanbesien/Updating_GT_t0_92X_dataRun2_HLT_v7_(9_3_X)

Updating GT for online (v7) and express (v6)

cmsbuild committed:
Merge pull request #20079 from bsunanda/Run2-alca89

Run2-alca89 Update the MIP calibration tool for HCAL

cmsbuild committed:
Merge pull request #19875 from nsmith-/layer1SecondStageLUT_pr93x

Move layer1SecondStageLUT to uint vector

cmsbuild committed:
Merge pull request #19905 from mrodozov/cmp-wngs-fix-l1trgr-regcalotrgr-gcc7

gcc7 compilation warning(s) fix(es) in L1Trigger/RegionalCaloTrigger

cmsbuild committed:
Merge pull request #19664 from davidlange6/clangTidyTest12-readability-static-definition-in-anonymous-namespace

clang check readability-static-definition-in-anonymous-namespace

cmsbuild committed:
Merge pull request #19680 from arizzi/overrideSplitMiniAOD

Override split level in miniaod

rmanzoni committed:
apply selections but isolation at the tau building step, sort taus by pt, fix isolation condition logic

kreczko committed:
Merge branch 'master' into kreczko/issue/19592

ggovi committed:
BuildFile for local compilation generalized

rmanzoni committed:
make produce abide to the promise of being const

davidlange6 committed:
Merge pull request #18575 from vkuznet/edmPickEvents-dasgo

Adopt code to use either das_client or dasgoclient

civanch committed:
fixed initialisation

rovere committed:
Add back ecalDrivenGsfElectrons

mmusich committed:
add TrackerMaps for SiStripDetVoff

smuzaffar committed:
added .clang-tidy

rovere committed:
Review3

* Improve python configuration files
* Switch to GEDPhotonProducer and make it more HGCal
  friendly.
* Remove globbing from event-content definitions

rmanzoni committed:
fix possible multiple file open

cmsbuild committed:
Merge pull request #20092 from ggovi/condcore-utilities-cond2xml-fix0-93X

Fix to extend conddb dump command to new CondFormats

cmsbuild committed:
Merge pull request #20085 from Martin-Grunewald/93XMoreL1TcompatWorkForTSG

More L1T compat work for TSG use (93X)

cmsbuild committed:
Merge pull request #20081 from makortel/trackingValidation_v10

Update tracking MC validation scripts

cmsbuild committed:
Merge pull request #20073 from makortel/fixTrackingOnly

Add built tracks back to trackingOnly MultiTrackValidator

cmsbuild committed:
Merge pull request #19971 from sikler/new-beamhlt_dqm_sourceclient_93X

new beamhlt dqm client for 93X (identical with 92X)

rmanzoni committed:
remove unneeded 'append' from dumpBeamSpotTxt, update AlCa

rmanzoni committed:
close file

kpedro88 committed:
update HCAL trigger mode for Run3

Sam-Harper committed:
adding all seeds for which pixelPairElectronSeeds are cleaned for to newCombinedSeeds

rovere committed:
Switch to GEDPhotonProducer

davidlange6 committed:
Merge pull request #20099 from smuzaffar/clang-tidy-file

added .clang-tidy

cmsbuild committed:
Merge pull request #20104 from kpedro88/Run3-WF2

update HCAL trigger mode for Run3

cmsbuild committed:
Merge pull request #20096 from civanch/fixed_stacking_action

Fixed initialisation of Geant4 stacking action

cmsbuild committed:
Merge pull request #20078 from DryRun/CMSSW_9_3_0_pre3_fix_hcalcalib

HCALDQM: Fix missing HF in hcalcalib

cmsbuild committed:
Merge pull request #20056 from vanbesien/Fixing_offline_HV_plot

Fixed range and unneeded spaces in histo labels

cmsbuild committed:
Merge pull request #20020 from ianna/dd-geom-clang-readability-container-size-empty

DetectorDescrition and Geometry Clang readability-container-size-empty

davidlange6 committed:
Merge pull request #19589 from bbockelm/change_flush_settings

Make compression settings more aggressive.

cmsbuild committed:
Merge pull request #19936 from PFCal-dev/hgc-tpg-integration-170727

HGCAL trigger updates (geometry, BH)

cmsbuild committed:
Merge pull request #19992 from davidcarbonis/phase1pixelValidation_930_fixes_v2

PixelPhase Validation bin reduction and protection against memory leak.

rmanzoni committed:
remove redundant check

deguio committed:
disable ZS mark and pass for 2017

makortel committed:
Remove QuadrupletSeedMerger as obsolete

davidlange6 committed:
Revert "Ss threadsafe ecal digi to raw"

fabferro committed:
adding patch to prevent corrupted row/col from crashing

fabferro committed:
removing extra ;

fabferro committed:
turning magic numbers into constants

davidlange6 committed:
Merge pull request #20119 from cms-sw/revert-18322-ss_threadsafe_EcalDigiToRaw

Revert "Ss threadsafe ecal digi to raw"

Commit:
fix issues reported by static analyzer

Commit:
improve code style

bbockelm committed:
Break up large read requests into smaller, pipelined requests.

This breaks up any read request over 8MB into a series of reads that
are 8MB or smaller.  In order to avoid network latency induced stalls,
we pipeline two requests at a time.

(cherry picked from commit 3e45943354bac92ee934301a4eb63ad07b5bf28f)

cmsbuild committed:
Merge pull request #20120 from CTPPS/ctpps_pixel_unpacker_patch1_93x

CTPPS Pixel unpacker patch

cmsbuild committed:
Merge pull request #20088 from rmanzoni/master_updatesHLTtau3mu

updates to tau3mu HLT producer

cmsbuild committed:
Merge pull request #20083 from zhangzc11/93X_dev_longLived_onlyDaughterToGeant4

for neutral exotic WIMPs - send their daughters to Geant4 instead of …

cmsbuild committed:
Merge pull request #20024 from mandrenguyen/playbackBugFix93X

Fix mixing workflows that use playback

cmsbuild committed:
Merge pull request #19939 from civanch/cleanup_g4_sensitive_detectors

Cleanup Geant4 sensitive detectors

rovere committed:
Review 4

* get rid of eras when not needed
* comply to CMSSW variable naming convention
* remove strongly typed parameters, when redefining in python
* remove globbing from output content definitions

makortel committed:
Move quadrupletseedmerging_cff under TkSeedingLayers and rename as PixelLayerQuadruplets_cff

makortel committed:
Add PixelLayerTriplets back

mmusich committed:
add calibration tree producer to unit tests and fix cms type declaration

ggovi committed:
Fixes for copy gt tool

makortel committed:
Fixes from the code style checker

ianna committed:
Clang readability-avoid-const-params-in-decls

bbockelm committed:
Fix code formatting issues.

(cherry picked from commit e468ae2df0219876168fe0185a840c7d65b1c012)

rovere committed:
Apply code-checks patch

bbockelm committed:
Rewrite logic around maintaining futures.

This collapses the two phases of read (create a list of work, then
create a list of futures) into a single while-loop that only needs
to maintain two futures (current and previous).

(cherry picked from commit b541b8d5e88c9aacea8e38e825846ff92a27e685)

rovere committed:
Add FromMultiCl gsf electrons to RECO and AOD

Commit:
Add a missing collection for Muons in HB/HE

fwyzard committed:
add warnings to L1TGlobalPrescaler

  - warn if `l1tPrescales` has the wrong number of elements
  - warn if the target prescales are lower than the original ones

cmsbuild committed:
Merge pull request #20103 from rmanzoni/master_fixBS

fix for Beam Spot txt file mess-up

cmsbuild committed:
Merge pull request #20129 from ianna/dd-readability-avoid-const-params-in-decls

DetectorDescription Clang readability-avoid-const-params-in-decls

fwyzard committed:
cleaned up by clang-tidy 4.0.0 modernize checks

Enabled checks:
    boost-use-to-string
    cppcoreguidelines-no-malloc
    google-build-explicit-make-pair
    misc-inefficient-algorithm
    misc-string-compare
    misc-uniqueptr-reset-release
    modernize-avoid-bind
    modernize-deprecated-headers
    modernize-loop-convert
    modernize-make-shared
    modernize-make-unique
    modernize-pass-by-value
    modernize-redundant-void-arg
    modernize-replace-auto-ptr
    modernize-shrink-to-fit
    modernize-use-auto
    modernize-use-bool-literals
    modernize-use-emplace
    modernize-use-equals-default
    modernize-use-equals-delete
    modernize-use-nullptr
    modernize-use-override
    modernize-use-transparent-functors
    modernize-use-using
    performance-faster-string-find
    performance-for-range-copy
    performance-implicit-cast-in-loop
    performance-inefficient-string-concatenation
    performance-type-promotion-in-math-fn
    performance-unnecessary-copy-initialization
    performance-unnecessary-value-param
    readability-container-size-empty
    readability-redundant-string-cstr
    readability-static-definition-in-anonymous-namespace
    readability-uniqueptr-delete-release

See http://releases.llvm.org/4.0.0/tools/clang/tools/extra/docs/clang-tidy/index.html for details.

Commit:
Update the cfg files

fwyzard committed:
minor improvements to the L1T and HLT monitoring modules

Commit:
Modify the BuildFile as suggested by code checker

cmsbuild committed:
Merge pull request #19978 from ianna/dd-clang-modernize-use-equals-delete

DetectorDescription Clang Check modernize-use-equals-delete

davidlange6 committed:
Merge pull request #20142 from bbockelm/pipeline_large_xrootd_reads_master

Pipeline large xrootd reads master

cmsbuild committed:
Merge pull request #20147 from fwyzard/HLTrigger_JSONMonitoring_minor_improvements_93x

minor improvements to the L1T and HLT monitoring modules

rovere committed:
Revert standard photon to PhotonProducer

rovere committed:
Revert "Revert standard photon to PhotonProducer"

This reverts commit 103c1866e7ed0911f63f26fc26ecaeb08cb4d3ee.

rovere committed:
Add parentSC to PhotonCoreProducer

thomreis committed:
Fix MonitorElement exception in L1TTau offlien DQM.

cmsbuild committed:
Merge pull request #20156 from thomreis/l1t_tau_offline_dqm_monitorelement_exception_fix

Fix MonitorElement exception in L1TTau offline DQM.

Martin-Grunewald committed:
Move @relval2017 default to HLT V2.2

cmsbuild committed:
Merge pull request #20144 from fwyzard/L1TGlobalPrescaler_add_warnings_93x

add warnings to L1TGlobalPrescaler (93x)

cmsbuild committed:
Merge pull request #19796 from kreczko/kreczko/issue/19592

[93X] Improvements to L1 Offline DQM - part 2 (issue #19592)

dildick commented:
@mulhearn @rekovic please sign

cmsbuild committed:
Merge pull request #20127 from makortel/removeQuadrupletSeedMerger

Remove QuadrupletSeedMerger as obsolete

cmsbuild committed:
Merge pull request #20159 from Martin-Grunewald/93Xrelval2017

Move @relval2017 default to HLT V2.2 (93X)

cmsbuild committed:
Merge pull request #20106 from Sam-Harper/EcalDrivenEffFix

Fixing missing seeds affecting Electron EcalDriven Efficiency

cmsbuild committed:
Merge pull request #20109 from mmusich/93X_addTkMapsSiStripDetVOff

add TrackerMaps for SiStripDetVOff Payload Inspector

cmsbuild committed:
Merge pull request #20115 from deguio/fixZS

disable ZS mark and pass for 2017 data. it affects pre-TS1 data only.

cmsbuild committed:
Merge pull request #20133 from ggovi/conddb-utilities-conddb-tool-fix3-93X

Fixes for conddb tool function to copy GT

cmsbuild committed:
Merge pull request #20138 from bsunanda/Run2-alca90

Run2-alca90 Add a missing collection for Muons in HB/HE

cmsbuild committed:
Merge pull request #20045 from rovere/HGCal_TDR

Add MultiCluster derived electrons and photons

Dr15Jones committed:
Added helper function convertToSingleModuleEndPaths

cmsbuild committed:
Merge pull request #20178 from Dr15Jones/addedConvertToSingleModuleEndPaths

Added helper function convertToSingleModuleEndPaths

cmsbuild committed:
Merge pull request #20139 from mmusich/93X_fixStripHitEffConfigAndTests

[93X] fix configuration of Strip hit efficiency tree and add tree producer to unit tests

cmsbuild committed:
Merge pull request #20012 from dildick/from-CMSSW_9_3_0_pre2-fix-issues-static-analyzer

Fix issues reported by static analyzer in CSCTriggerPrimitives

dildick commented:
Ill resolve the merge conflicts asap

Commit:
solve merge conflicts

cmsbuild commented:
Pull request #19778 was updated. @ghellwig, @lveldere, @perrozzi, @ianna, @kpedro88, @fabozzi, @Martin-Grunewald, @mommsen, @vanbesien, @govoni, @perrotta, @civanch, @monttj, @silviodonato, @cmsbuild, @fwyzard, @GurpreetSinghChahal, @davidlange6, @smuzaffar, @Dr15Jones, @kkousour, @ssekmen, @mdhildreth, @cerminar, @slava77, @ggovi, @mulhearn, @efeyazgan, @vazzolini, @rekovic, @kmaeshima, @arunhep, @emeschi, @thuer, @dmitrijus, @alja, @franzoni, @lpernie can you please check and sign again.

dildick commented:
need to rebase

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-19778/169
